lib/framedivider.rb

```diff
@@ -5,94 +5,216 @@
 #
 # @param [Array] pts OpenStudio Point3D array/vector
 #
-# @return [Array] Flattened OS 3D points array
+# @return [Array] Flattened OpenStudio 3D points array
 def flatZ(pts)
   unless pts && (pts.is_a?(OpenStudio::Point3dVector) || pts.is_a?(Array))
-    TBD.log(TBD::DEBUG,
-      "Invalid pts to flatten (argument) - skipping")
-    return pts
+    TBD.log(TBD::DEBUG, "Invalid pts to flatten (argument) - skipping")
+    return OpenStudio::Point3dVector.new
   end
 
+  vec = OpenStudio::Point3dVector.new
   pts.each do |pt|
     unless pt.class == OpenStudio::Point3d
-      TBD.log(TBD::DEBUG,
-        "#{pt.class}? expected OSM 3D points to flatten - skipping")
-      next
+      msg = "#{pt.class}? expected OSM 3D points to flatten - skipping"
+      TBD.log(TBD::DEBUG, msg)
+      return OpenStudio::Point3dVector.new
     end
-    pt = OpenStudio::Point3d.new(pt.x, pt.y, 0)
+    vec << OpenStudio::Point3d.new(pt.x, pt.y, 0)
   end
-  pts
+  vec
 end
 
-##
-# Validate whether an OpenStudio subsurface can 'fit in' (vs parent, siblings)
+## Validates whether one OpenStudio polygon fits within another polygon.
 #
-# @param [OpenStudio::Model::Model] model An OS model
-# @param [String] id SubSurface identifier
-# @param [Array] pts Point3D array of subsurface 'id'
+# @param [Array] poly1 Point3D array of convex polygon #1
+# @param [Array] poly1 Point3D array of convex polygon #2
+# @param [String] id1 Polygon #1 identifier (optional)
+# @param [String] id2 Polygon #2 identifier (optional)
 #
-# @return [Bool] Returns true if subsurface can fit in.
-def fit?(model, id, pts)
-  unless model && model.is_a?(OpenStudio::Model::Model)
-    TBD.log(TBD::DEBUG,
-      "Can't find or validate OSM (argument) for subsurface fit - skipping")
+# @return Returns true if either polygon fits entirely within the other.
```

Comment:
Is it allowed to share one or more edges?  What about all edges?

Reply:
Yes. There's a series of tests [here](https://github.com/rd2/tbd/blob/86bde630e3331f133f55097e25017cea2e11f96f/spec/tbd_tests_spec.rb#L4670).